bigtable/bigtable-proxy/src/main/java/com/google/cloud/bigtable/examples/proxy/commands/Serve.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigtable.examples.proxy.commands;
+
+import com.google.bigtable.admin.v2.BigtableInstanceAdminGrpc;
+import com.google.bigtable.admin.v2.BigtableTableAdminGrpc;
+import com.google.bigtable.v2.BigtableGrpc;
+import com.google.cloud.bigtable.examples.proxy.core.ProxyHandler;
+import com.google.cloud.bigtable.examples.proxy.core.Registry;
+import com.google.common.collect.ImmutableMap;
+import com.google.longrunning.OperationsGrpc;
+import io.grpc.InsecureServerCredentials;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Server;
+import io.grpc.ServerCallHandler;
+import io.grpc.netty.shaded.io.grpc.netty.NettyServerBuilder;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Help.Visibility;
+import picocli.CommandLine.Option;
+
+@Command(name = "serve", mixinStandardHelpOptions = true, description = "Start the proxy server")
+public class Serve implements Callable<Void> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(Serve.class);
+
+  @Option(
+      names = "--listen-port",
+      required = true,
+      description = "Local port to accept connections on")
+  int listenPort;
+
+  @Option(names = "--useragent", showDefaultValue = Visibility.ALWAYS)
```

Comment:
will this override the userAgent that's set on the underlying client?

Reply:
I meant when we send request through the proxy:

some client -> proxy -> server

will the user agent from "some client" get passed?